wallet/walletrpc.c

```diff
@@ -404,17 +407,24 @@ static struct command_result *json_prepare_tx(struct command *cmd,
 	/* Add the change as the last output */
 	if (!amount_sat_eq((*utx)->wtx->change, AMOUNT_SAT(0))) {
 		struct bitcoin_tx_output *change_output;
+		struct amount_sat change;
 
 		changekey = tal(tmpctx, struct pubkey);
 		if (!bip32_pubkey(cmd->ld->wallet->bip32_base, changekey,
 				  (*utx)->wtx->change_key_index))
 			return command_fail(cmd, LIGHTNINGD, "Keys generation failure");
 
-		change_output = new_tx_output(outputs, (*utx)->wtx->change,
+
+		if (zero_out_change && *zero_out_change)
```

Comment:
zero_out_change will be non-NULL, since you used p_opt_def().

Reply:
nice.